src/Common/src/System/Data/Common/FieldNameLookup.cs

```diff
@@ -12,142 +12,38 @@
 
 namespace System.Data.ProviderBase
 {
-    internal sealed class FieldNameLookup
+    internal sealed class FieldNameLookup : BasicFieldNameLookup
     {
-        // Dictionary stores the index into the _fieldNames, match via case-sensitive
-        private Dictionary<string, int> _fieldNameLookup;
-
-        // original names for linear searches when exact matches fail
-        private string[] _fieldNames;
-
-        // if _defaultLocaleID is -1 then _compareInfo is initialized with InvariantCulture CompareInfo
-        // otherwise it is specified by the server for the correct compare info
-        private CompareInfo _compareInfo;
         private int _defaultLocaleID;
 
-        public FieldNameLookup(string[] fieldNames, int defaultLocaleID)
+        public FieldNameLookup(string[] fieldNames, int defaultLocaleID) : base(fieldNames)
         {
-            if (null == fieldNames)
-            {
-                throw ADP.ArgumentNull(nameof(fieldNames));
-            }
-            _fieldNames = fieldNames;
             _defaultLocaleID = defaultLocaleID;
         }
 
-        public FieldNameLookup(System.Collections.ObjectModel.ReadOnlyCollection<string> columnNames, int defaultLocaleID)
+        public FieldNameLookup(System.Collections.ObjectModel.ReadOnlyCollection<string> columnNames, int defaultLocaleID) : base(columnNames)
         {
-            int length = columnNames.Count;
-            string[] fieldNames = new string[length];
-            for (int i = 0; i < length; ++i)
-            {
-                fieldNames[i] = columnNames[i];
-            }
-            _fieldNames = fieldNames;
             _defaultLocaleID = defaultLocaleID;
-            GenerateLookup();
         }
 
-        public FieldNameLookup(DbDataReader reader, int defaultLocaleID)
+        public FieldNameLookup(DbDataReader reader, int defaultLocaleID) : base(reader)
         {
-            int length = reader.FieldCount;
-            string[] fieldNames = new string[length];
-            for (int i = 0; i < length; ++i)
-            {
-                fieldNames[i] = reader.GetName(i);
-            }
-            _fieldNames = fieldNames;
             _defaultLocaleID = defaultLocaleID;
         }
 
-        public int GetOrdinal(string fieldName)
-        {
-            if (null == fieldName)
-            {
-                throw ADP.ArgumentNull(nameof(fieldName));
-            }
-            int index = IndexOf(fieldName);
-            if (-1 == index)
-            {
-                throw ADP.IndexOutOfRange(fieldName);
-            }
-            return index;
-        }
-
-        public int IndexOfName(string fieldName)
+        //The compare info is specified by the server by specifying the default LocaleId.
+        new CompareInfo GetCompareInfo()
```

Comment:
I'm not sure this is going to behave the way you want it to...

I think in the base class, you want to make the method `virtual`, e.g.:

```csharp
protected virtual CompareInfo GetCompareInfo() => CultureInfo.InvariantCulture.CompareInfo;
```

Then here in the subclass, override the virtual method:

```csharp
protected override CompareInfo GetCompareInfo()
{
    ...
}
```